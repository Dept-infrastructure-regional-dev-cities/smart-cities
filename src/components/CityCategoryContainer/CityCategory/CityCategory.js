import React from 'react';
import PropTypes from 'prop-types';
import PageWrapper from '../../PageWrapper/PageWrapper';
import PageBanner from '../../PageBanner/PageBanner';
import SubCategorySummary from '../../SubCategorySummary/SubCategorySummary';
import style from './CityCategory.scss';

const CityCategory = (props) => {
  const title = `I am the "${props.category.name}" page for "${props.city.name}"`;

  return (
    <PageWrapper
      cityId={props.city.id}
      categoryId={props.category.id}
      cityName={props.city.name}
      categoryColorName={props.category.colorName}
    >
      <PageBanner
        colorName={props.category.colorName}
        description={props.category.description}
        indicator={props.category.heroIndicatorId}
        title={props.category.name}
        cities={props.cities}
        isCategoryPage
      />

      {props.category.subCategories.map(subCategory => (
        <SubCategorySummary
          key={subCategory.name}
          {...subCategory}
          colorName={props.category.colorName}
          city={props.city}
        />
      ))}

      <div className={style.main}>
        <h1>{title}</h1>

        <pre>{JSON.stringify(props.category, null, 2)}</pre>
        <pre>{JSON.stringify(props.cities, null, 2)}</pre>
      </div>
    </PageWrapper>
  );
};

CityCategory.propTypes = {
  category: PropTypes.shape({
    colorName: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    heroIndicatorId: PropTypes.string.isRequired,
    subCategories: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      indicatorIds: PropTypes.arrayOf(PropTypes.string).isRequired,
    })).isRequired,
  }).isRequired,
  city: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  }).isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  })).isRequired,
  toggleCitySelected: PropTypes.func.isRequired,
};

export default CityCategory;
