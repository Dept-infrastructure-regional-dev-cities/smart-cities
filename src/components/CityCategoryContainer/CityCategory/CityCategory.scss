@import '~style/variables';

.main {
  @include container;
}
