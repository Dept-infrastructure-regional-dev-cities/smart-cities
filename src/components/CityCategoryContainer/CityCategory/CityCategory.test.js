import React from 'react';
import { shallow } from 'enzyme';
import CityCategory from './CityCategory';

const defaultProps = {
  category: {
    name: 'Mock category',
    id: 'mock',
  },
  city: {
    name: 'Perth',
    id: 'perth',
  },
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityCategory {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
