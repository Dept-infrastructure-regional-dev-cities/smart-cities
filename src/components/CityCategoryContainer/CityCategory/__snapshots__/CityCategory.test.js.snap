// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper cityId=\\"perth\\" categoryId=\\"mock\\" cityName=\\"Perth\\" categoryColorName=\\"jobs\\">
  <PageBanner colorName=\\"jobs\\" description=\\"The description of the category\\" indicator=\\"population\\" title=\\"Mock category\\" cities={{...}} isCategoryPage={true} />
  <SubCategorySummary name=\\"Buying houses\\" colorVariation=\\"100\\" indicatorIds={{...}} colorName=\\"jobs\\" city={{...}} />
  <SubCategorySummary name=\\"Living in houses\\" colorVariation=\\"050\\" indicatorIds={{...}} colorName=\\"jobs\\" city={{...}} />
  <div className=\\"main\\">
    <h1>
      I am the &quot;Mock category&quot; page for &quot;Perth&quot;
    </h1>
    <pre>
      {
        &quot;name&quot;: &quot;Mock category&quot;,
        &quot;description&quot;: &quot;The description of the category&quot;,
        &quot;colorName&quot;: &quot;jobs&quot;,
        &quot;heroIndicatorId&quot;: &quot;population&quot;,
        &quot;id&quot;: &quot;mock&quot;,
        &quot;subCategories&quot;: [
          {
            &quot;name&quot;: &quot;Buying houses&quot;,
            &quot;colorVariation&quot;: &quot;100&quot;,
            &quot;indicatorIds&quot;: []
          },
          {
            &quot;name&quot;: &quot;Living in houses&quot;,
            &quot;colorVariation&quot;: &quot;050&quot;,
            &quot;indicatorIds&quot;: []
          }
        ]
      }
    </pre>
    <pre>
      [
        {
          &quot;id&quot;: &quot;perth&quot;,
          &quot;description&quot;: &quot;description of Perth&quot;,
          &quot;name&quot;: &quot;Perth&quot;
        },
        {
          &quot;id&quot;: &quot;sydney&quot;,
          &quot;description&quot;: &quot;description of Sydney&quot;,
          &quot;name&quot;: &quot;Sydney&quot;
        },
        {
          &quot;id&quot;: &quot;melbourne&quot;,
          &quot;description&quot;: &quot;description of Melbourne&quot;,
          &quot;name&quot;: &quot;Melbourne&quot;
        }
      ]
    </pre>
  </div>
</PageWrapper>"
`;
